packages/core/src/driver.ts

```diff
@@ -96,7 +96,110 @@ interface DriverConfig {
   trust?: TrustStrategy
   fetchSize?: number
   logging?: LoggingConfig
+}
+
+/**
+ * The session configuration
+ *
+ * @interface
+ *
+ * @param {string} defaultAccessMode=WRITE - The access mode of this session, allowed values are {@link READ} and {@link WRITE}.
+ * @param {string|string[]} bookmarks - The initial reference or references to some previous
+ * transactions. Value is optional and absence indicates that that the bookmarks do not exist or are unknown.
+ * @param {number} fetchSize - The record fetch size of each batch of this session.
+ * Use {@link FETCH_ALL} to always pull all records in one batch. This will override the config value set on driver config.
+ * @param {string} database - The database this session will operate on.
+ * @param {string} impersonatedUser - The username which the user wants to impersonate for the duration of the session.
+ * @param {BookmarkManager} [bookmarkManager] = The bookmark manager
+ */
+class SessionConfig {
```

Comment:
isn't going from a raw object to a class a breaking change? (which is OK for 5.0 anyway, just wondering)

Reply:
Nope, typescript has loose type. This class will behave an interface actually, I need to define as class to make it possible add the experimental annotations in place. 

See usage example: https://github.com/neo4j/neo4j-javascript-driver/pull/974/files#diff-0ec3f1aeef14ced21b829fd32b84fb3a078e728e5fe474d34b33912dee2b7a1bR124